sunshine/confighttp.cpp

```diff
@@ -412,6 +417,67 @@ void deleteApp(resp_https_t response, req_https_t request) {
   proc::refresh(config::stream.file_apps);
 }
 
+void uploadCover(resp_https_t response, req_https_t request) {
+  if(!authenticate(response, request)) return;
+
+  std::stringstream ss;
+  std::stringstream configStream;
+  ss << request->content.rdbuf();
+  pt::ptree outputTree;
+  auto g = util::fail_guard([&]() {
+    std::ostringstream data;
+
+    SimpleWeb::StatusCode code = SimpleWeb::StatusCode::success_ok;
+    if(outputTree.get_child_optional("error").has_value()) {
+      code = SimpleWeb::StatusCode::client_error_bad_request;
+    }
+
+    pt::write_json(data, outputTree);
+    response->write(code, data.str());
+  });
+  pt::ptree inputTree;
+  try {
+    pt::read_json(ss, inputTree);
+  }
+  catch(std::exception &e) {
+    BOOST_LOG(warning) << "UploadCover: "sv << e.what();
+    outputTree.put("status", "false");
+    outputTree.put("error", e.what());
+    return;
+  }
+
+  auto key = inputTree.get("key", "");
+  if(key.empty()) {
+    outputTree.put("error", "Cover key is required");
+    return;
+  }
+  auto url = inputTree.get("url", "");
+
+  const std::string coverdir = SUNSHINE_ASSETS_DIR "/covers/";
```

Comment:
Can we use this?

```cpp
const std::string coverdir = platf::appdata().string() + "/covers/"
```

Since the `SUNSHINE_ASSETS_DIR` will get overwritten on updates normally.

Reply:
This sounds more reasonable.